Core/CoreTests/CoreTestCase.swift

```diff
@@ -51,14 +51,21 @@ class CoreTestCase: XCTestCase {
         router = TestRouter()
         logger = TestLogger()
         TestsFoundation.singleSharedTestDatabase = resetSingleSharedTestDatabase()
-        environment = AppEnvironment.shared
+        environment = TestEnvironment()
```

Comment:
In tests can we update AppEnvironment.shared instead of passing around a different one? There are some places where we can't (or would rather not) plumb in a specific AppEnvironment.